solidity/ecdsa/deploy/09_transfer_proxy_admin_ownership.ts

```diff
@@ -0,0 +1,24 @@
+import type { ProxyAdmin } from "../typechain"
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
+  const { ethers, getNamedAccounts, upgrades, deployments } = hre
+  const { esdm } = await getNamedAccounts()
+  const { deployer } = await ethers.getNamedSigners()
+
+  deployments.log(`transferring ProxyAdmin ownership to ${esdm}`)
+
+  // TODO: Once a DAO is established we want to switch to ProxyAdminWithDeputy and
+  // use the DAO as the proxy admin owner and ESDM as the deputy. Until then we
+  // use ESDM as the owner of ProxyAdmin contract.
+  const newProxyAdminOwner = esdm
```

Comment:
@pdyraga is this correct? Who should be responsible for upgrading the WalletRegistry contract?

Reply:
We need to discuss it with the community but I would like to avoid transferring the ownership to DAO until the system is fully launched and confirmed to be stable. In the first weeks or months, this has to be developer-owned. I think it's fine to assume this will be ESDM initially.

Reply:
Reading it one more time, yes - the comment in TODO is correct. By the time we deploy tBTC v2 the DAO should be fully deployed including [`ProxyAdminWithDeputy`](https://github.com/threshold-network/solidity-contracts/blob/main/contracts/governance/ProxyAdminWithDeputy.sol). `ProxyAdminWithDeputy` should give us the functionalities we need.